File: src/main/java/com/server/ordering/domain/PhoneNumber.java
package com.server.ordering.domain;

import lombok.*;

import javax.persistence.*;

import static javax.persistence.GenerationType.*;
import static lombok.AccessLevel.*;

@Entity
@Getter
@NoArgsConstructor(access = PROTECTED)
@RequiredArgsConstructor
public class PhoneNumber {

    @Id @GeneratedValue(strategy = IDENTITY)
    @Column(name = "phone_number_id")
    private Long id;

    @NonNull
    private String phoneNumber;

    @NonNull
    @Enumerated(value = EnumType.STRING)
    private MemberType memberType;

    public void putPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/PhoneNumber.kt
package com.byeoru.ordering_server.domain

import javax.persistence.*

@Entity
class PhoneNumber(
    phoneNumber: String,
    memberType: MemberType) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "phone_number_id")
     var id: Long? = null
         protected set

     var phoneNumber: String = phoneNumber
         protected set

    @Enumerated(value = EnumType.STRING)
     var memberType: MemberType = memberType
         protected set

    fun putPhoneNumber(phoneNumber: String) {
        this.phoneNumber = phoneNumber
    }
}